storm/src/main/storm/mesos/MesosCommon.java

```diff
@@ -85,9 +92,29 @@ public static boolean startLogViewer(Map conf) {
   }
 
   public static int portFromTaskId(String taskId) {
-    int last = taskId.lastIndexOf("-");
-    String port = taskId.substring(last + 1);
-    return Integer.parseInt(port);
+    String[] parts = taskId.trim().split("-");
+    if (parts.length < 3) {
+      throw new IllegalArgumentException("TaskID " + taskId.trim() + " is invalid. " +
+          "Number of dash-delimited components (" + parts.length + ") is less than expected. " +
+          "Expected format is HOSTNAME-PORT-TIMESTAMP");
+    }
+
+    // TaskID format: HOSTNAME-PORT-TIMESTAMP. Notably, HOSTNAME can have dashes too,
+    // so the port is the 2nd-to-last part after splitting on dash.
+    String portString = parts[parts.length - 2];
```

Comment:
@JessicaLHartog Not sure what you mean - I was just proposing not to do a subtraction.